src/twisted/internet/defer.py

```diff
@@ -160,6 +166,65 @@ def maybeDeferred(f, *args, **kw):
 
 
 
+def unbox(deferred, raises=True, passthrough=True):
```

Comment:
I think that choosing a good name here is really, really important.

"unbox" is a bit jargon-y, and is only halfway accurate in this case.  Maybe something like `extractImmediateResult()`?

The docstring should also explain the one-and-only valid use-case for this function.  As @exarkun noted, lots of people want this function, except what they _actually_ want is "spin the reactor and block until a result is available", which is something we cannot do.